lib/cretonne/src/lib.rs

```diff
@@ -46,3 +58,22 @@ mod stack_layout;
 mod topo_order;
 mod unreachable_code;
 mod write;
+
+/// This replaces `std` in builds with no_std.
+#[cfg(feature = "no_std")]
+mod std {
+    pub use core::*;
+    #[macro_use]
+    pub use alloc::{boxed, vec, string};
+    pub mod prelude {
+        pub use core::prelude as v1;
```

Comment:
I don't see `v1` used anywhere; is this line needed?

Reply:
Not sure why I put that there. I think I saw that `nom` (the parser) had that in their version of an impromptu `std` and thought that having a prelude would make sense. Since it isn't automatically included in every file, obviously it's not. 😞